### PR TITLE
Fix for issue 117, scope for variadic is punctuation.definition.variable.php

### PR DESCRIPTION
**Pull request by zr9**

### CODE REVIEW

File: grammars/php.cson [hunk 1]
```diff
@@ -361,7 +361,7 @@
...
               '7':
                 'name': 'invalid.illegal.non-null-typehinted.php'
-            'match': '(?xi)\n        \t\t\t\t\t\t\t\\s*([a-z_][a-z_0-9]*)? # Typehinted class name\n        \t\t\t\t\t\t\t\\s*(&)? \t\t\t\t\t# Reference\n        \t\t\t\t\t\t\t\\s*((\\$+)[a-z_\\x{7f}-\\x{ff}][a-z0-9_\\x{7f}-\\x{ff}]*) # The variable name\n        \t\t\t\t\t\t\t(?:\n        \t\t\t\t\t\t\t\t\\s*(?:(=)\\s*(?:(null)|((?:\\S*?\\(\\))|(?:\\S*?))))\t# A default value\n        \t\t\t\t\t\t\t)?\n        \t\t\t\t\t\t\t\\s*(?=,|\\)|/[/*]|\\#|$) # A closing parentheses (end of argument list) or a comma\n\t\t\t\t\t        '
+            'match': '(?xi)\n        \t\t\t\t\t\t\t\\s*([a-z_][a-z_0-9]*)? # Typehinted class name\n        \t\t\t\t\t\t\t\\s*(&)? \t\t\t\t\t# Reference\n        \t\t\t\t\t\t\t\\s*(((?:\\.\\.\\.)?\\$+)[a-z_\\x{7f}-\\x{ff}][a-z0-9_\\x{7f}-\\x{ff}]*) # The variable name\n        \t\t\t\t\t\t\t(?:\n        \t\t\t\t\t\t\t\t\\s*(?:(=)\\s*(?:(null)|((?:\\S*?\\(\\))|(?:\\S*?))))\t# A default value\n        \t\t\t\t\t\t\t)?\n        \t\t\t\t\t\t\t\\s*(?=,|\\)|/[/*]|\\#|$) # A closing parentheses (end of argument list) or a comma\n\t\t\t\t\t        '
```
Comment (50Wliu) [inline grammars/php.cson]: This should also have its whitespace honored

**Comment by 50Wliu** [comment]
Same as #119: this needs specs.

**Comment by 50Wliu** [comment]
Can you also add a test for `meta.function.argument.typehinted.php`?  Thanks!

**Comment by zr9** [comment]
Done :) I'm updated scope as well, so it now "punctuation.definition.variable.php" same as for "$" i'm think it more correct(it was other for no-default). And spec test is contains typehinted class test as well since there was no test for that.

**Comment by 50Wliu** [comment]
Looks good to me!  If no one objects, I'll merge it sometime tomorrow or Wednesday.

**Comment by zr9** [comment]
@50Wliu Could you merge please so i'm can clean whole file? Seems like no one objects

**Comment by 50Wliu** [comment]
Thanks @zr9!

**Comment by zr9** [comment]
@50Wliu You're welcome :) I'm also added fix in CSS highlight, so if have a time you can review it as well. I'm in the meantime start cleaning this one from garbage symbols

**Pull request merged**